test/code/plugins/oms_common_test.rb

```diff
@@ -94,6 +100,148 @@ def teardown
       "OSAlias=UniversalR\n" \
       "OSManufacturer=Central Logistics GmbH\n"
 
+    # Begin:  Tests on Public internals; probably only for use inside the Singular object.
+
+    def test_clean_hostname_string
+      hostname = '        microsoft     '
+      b = Common.clean_hostname_string(hostname)
+      expected = 'microsoft'
+      assert_equal(b, expected, "method clean_hostname_string should return '#{expected}'.")
+      hostname = expected
+      b = Common.clean_hostname_string(hostname)
+      assert_equal(b, expected, "method clean_hostname_string should return '#{expected}'.")
+    end
+
+    def test_has_designated_hostnamefile?
+      file_hostname = '/tmp/containerhostname'
```

Comment:
Again, please use the Ruby Tempfile module, otherwise this can collide with other users on shared build machines!

Reply:
Sorry.  Missed that first time through.